Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2432,12 +2432,37 @@ extension BrowserViewController: WKNavigationDelegate {
         // instead of being loaded in the webview. Note that there is no point in calling canOpenURL() here, because
         // iOS will always say yes. TODO Is this the same as isWhitelisted?
 
-        if isAppleMapsURL(url) || isStoreURL(url) {
-            UIApplication.shared.openURL(url)
+       if isAppleMapsURL(url) {
+           UIApplication.shared.openURL(url)
+           decisionHandler(WKNavigationActionPolicy.cancel)
+           return
+       }
+
+        if let tab = tabManager.selectedTab, isStoreURL(url) {
             decisionHandler(WKNavigationActionPolicy.cancel)
+
+            let alreadyShowingSnackbarOnThisTab = tab.bars.count > 0
+            if alreadyShowingSnackbarOnThisTab {
+                return
+            }
+            
+            let msg =  NSAttributedString(string: Strings.ExternalLinkToAppStore_ConfirmationTitle, attributes: [NSForegroundColorAttributeName: UIConstants.TableViewRowTextColor])
+            
+            let snackBar = TimerSnackBar(attrText: msg,
```

Comment:
I think it would be better if we removed this from BVC to keep this method more readable. 

If you added a static func to TimedSnackBar to create the bar and then simply use it in this method. 
something like
`let snackbar = TimedSnackBar.createAppStoreBar(url, tab)`
`tab.addsnackbar(snackbar)`